### DISCUSSION TITLE
Invalid file signature

### DISCUSSION THREAD

**Discussion by hado010** [arendst/Tasmota#10384] [Q&A] [answered]
Hello,

Im trying to put a Firmware Upgrade on my Zigbee Bridge V1.1 2020.02.28.
Within Tasmota, via option Upgrade by file upload I choose the file: ncp-uart-sw_6.7.6_115200.ota or ncp-uart-sw_6.5.5_115200.ota.
In both cases I get the message: Upload Failed, Invalid file signature.

What do I do wrong?

Thanks

![foutmelding](https://user-images.githubusercontent.com/76902371/103476894-d2736b00-4db9-11eb-860d-3f03b62944bd.jpg)

**Comment by s-hadinger** [comment, marked as answer]
Either you didn't download correctly the file, or you didn't select Module 75. 

What is the length in bytes of the firmware yiu are trying to upload?

**Reply by hado010** [reply]
Hi, thanks for your reaction.
This is the file:
ncp-uart-sw_6.5.5_115200.ota = 177 kB (181.500 bytes)
or
ncp-uart-sw_6.7.6_115200.ota = 185 kB (190.272 bytes)

Rgrds

**Reply by s-hadinger** [reply]
Sizebis correct. Can you please send the result of `module` command?

**Reply by hado010** [reply]
19:40:33 CMD: module
19:40:33 SRC: WebConsole from 192.xxx.xxx.xxx
19:40:33 CMD: Group 0, Index 1, Command "MODULE", Data ""
19:40:33 RSL: stat/tasmota_zbBridge2/RESULT = {"Module":{"0":"Sonoff ZHABridge"}}

**Reply by s-hadinger** [reply]
As said before, you must use `Module 75` to flash the Zigbee MCU. What you tried here is to flash the Zigbee firmware to Tasmota. 

hat guide are you using? It is either wrong or you are not following it.

**Reply by trichromic-ab** [reply]
I'm in the same situation with 6.7.6 and 6.7.8. 
Could you explain a little more about what you mean by "select module 75" ?
I'm new to this technology.

**Comment by hado010** [comment]
Thank you for you "Module 75" tip! Now I am one step futher:
![successful](https://user-images.githubusercontent.com/76902371/104102757-a024aa80-529e-11eb-8a5b-2199ee29ebb8.jpg)

**Reply by hado010** [reply]
@s-hadinger, Thanks a lot for your help! I now have the Bridge working with Tasmota within HomeAssistant, great!

**Reply by trichromic-ab** [reply]
Hi, could you explain what you did to select Module 75? 
I'm having the same "invalid file signature" problem.

**Reply by sfromis** [reply]
You can enter `Module 75` as a command in the console. Or you can find `Sonoff ZbBridge (75)` in the dropdown in configure module or configure template.

**Comment by trichromic-ab** [comment]
Hi and thanks for getting back so quickly.
Actually after a few hours of pondering, I realised I had flashed with the basic tasmota firmware (which tasmotizer selects by default) and there is no module 75 in that.
So then I went to flash the zbridge version and now progress made at last.

**Reply by WellThatIsntRight** [reply]
this was the key piece of logic i was missing! thanks

**Reply by trichromic-ab** [reply]
Welcome!

**Comment by jactora** [comment]
can somebody please help me. i have same signature issue. i ran module cmd and here is the result
1:31:59.572 CMD: module
21:31:59.575 SRC: WebConsole from 192.168.31.179
21:31:59.578 CMD: Group 0, Index 1, Command "MODULE", Data ""
21:31:59.583 RSL: stat/tasmota_04D02A/RESULT = {"Module":{"0":"Flash ZBBridge"}}

Program Version | 9.3.1(tasmota)
Build Date & Time | 2021-03-09T16:12:28
Core/SDK Version | 2_7_4_9/2.2.2-dev(38a443e)

one mistake i did was instead of uploading firmware, i did upgrade by web server first. it went fine.
but then checked the instructions, i was supposed to use upgrade by file upload. did i break anything?

**Reply by sfromis** [reply]
Upgrade by web server vs file upload is just the same, it is only a practical question of you having it on a http server or not. You still need to flash the correct firmware binary, tasmota-zbbridge.bin instead of tasmota.bin. You can do OTA upgrade to http://ota.tasmota.com/tasmota/release/tasmota-zbbridge.bin.gz

**Reply by jactora** [reply]
thanks. i used the url above and ran upgrade. all went fine. bridge restarted.
but in the console logs, i dont see any zigbee logs.
and i cannot connect in HA via ZHA

**Reply by sfromis** [reply]
If you run ZHA, Tasmota is out of the loop, *everything* must be handled in ZHA, do not even expect any activity in the Tasmota log.

**Reply by jactora** [reply]
sorry what i meant was - i do not see the Tasmota Zigbee message of Zigbee started in the console.

**Reply by jactora** [reply]
thanks got sorted after few tries

**Comment by avviano** [comment]
I have the same issue, but I did select 75. I always get the message Invalid file signature. Here is the log message: 

09:40:14.434 UPP: Multicast (re)joined
09:40:14.756 HTP: Upload done
09:40:15.724 HTP: Main Menu
09:40:17.085 HTP: Firmware Upgrade
09:40:24.209 ZbEZSPReceived: time-out, discarding FE0B058A0286980109006E
09:40:25.821 UPP: Multicast disabled
09:40:25.871 CFG: Saved to flash at 1FB, Count 35, Bytes 4096
09:40:25.872 UPL: File ncp-uart-sw_6.7.8_115200.ota
09:40:25.892 UPP: Multicast (re)joined
09:40:26.223 HTP: Upload done
09:41:32.513 ZbEZSPReceived: time-out, discarding FE0B058A02463901710077

09:46:10.318 CMD: module cmd
09:46:10.319 SRC: WebConsole from 192.168.178.39
09:46:10.320 CMD: Grp 0, Cmnd 'MODULE', Idx 1, Len 3, Data 'cmd'
09:46:10.322 RSL: RESULT = {"Module":{"75":"Sonoff ZbBridge"}}

**Reply by barbudor** [reply]
What is the size in bytes of your ncp-uart-*.ota file ?

**Comment by avviano** [comment]
Version 6.7.6 = 124 KB
Version 6.7.8 = 128 KB
Version 6.7.9 = 128 KB

In Bytes:

Version 6.7.6 = 126.107 Bytes
Version 6.7.8 = 130.508 Bytes
Version 6.7.9 = 130.520 Bytes

**Reply by barbudor** [reply]
They are all wrong. It should be around 187KB
You haven't downloaded the raw binary file but Github's HTML page

You shouldn't right click on the file name but click normally on the name, then click on the "Download" button on the right
![image](https://user-images.githubusercontent.com/2996042/124258918-3d65ea80-db2e-11eb-8a8f-ee5ad7db280e.png)

**Comment by avviano** [comment]
What a newbie mistake to make.  Thank you for your patience and bearing with me! 

I now re-downloaded and it worked. I am glad I did post here, I was about to give up.

Thanks again, made my day!!!

**Comment by NoGrout** [comment]
Hello, I have a sonoff zbridge pro, I can't select module 75. It is missing.
The update is therefore invalid as on this post.
Could you help me ? Thanks

**Reply by sfromis** [reply]
As the Sonoff ZB Bridge-P is no longer ESP8266, but the improved ESP32, module 75 no longer applies, that is for the older non-"pro" device. You should follow the installation instructions for your device, which will come up with the template configured to match the hardware, without you needing to select anything for that.

**Reply by NoGrout** [reply]
Thank you for your help.  
I understand this information. 
 But I have this same type of error.  
 
the tutorial from https://notenoughtech.com/home-automation/tasmota-on-sonoff-zb-bridge-pro/ is completely validated without error.  

But my zigbee panel remains greyed out.  the zigbee version is not displayed. 

However, it has already worked once.  I've been stuck on this for several days.
Thank you for help

**Reply by sfromis** [reply]
When you just write things like "this same type of error", you are keeping your symptoms a secret, unavailable to guesses about what's missing. In general, pertinent details from the log are useful.

**Reply by sfromis** [reply]
Anyway, the "Not Enough Tech" guide is really outdated, if you start by flashing the specific image for the device, you can skip most of the text, but instead start at "Flashing coordinator firmware on Sonoff ZB Bridge Pro".

**Reply by NoGrout** [reply]
I will create a post with all my logs.  It will be easier for tracking.
Thank you 

Edit: Hello, I put the details here.  thank you

https://github.com/arendst/Tasmota/discussions/17321
